File: solution/2_SchrijfRomeinsGetalInArabischeCijfers/2_SchrijfRomeinsGetalInArabischeCijfers/SchrijfInArabischeCijfersTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using num;

namespace _2_SchrijfRomeinsGetalInArabischeCijfers
{
    [TestClass]
    public class SchrijfInArabischeCijfersTest
    {
        [TestMethod]
        public void individueleRomeinseGetallen() {
		    Assert.AreEqual(1, GetalUtil.alsArabischCijfer("I"));
		    Assert.AreEqual(5, GetalUtil.alsArabischCijfer("V"));
		    Assert.AreEqual(10, GetalUtil.alsArabischCijfer("X"));
		    Assert.AreEqual(50, GetalUtil.alsArabischCijfer("L"));
		    Assert.AreEqual(100, GetalUtil.alsArabischCijfer("C"));
		    Assert.AreEqual(500, GetalUtil.alsArabischCijfer("D"));
		    Assert.AreEqual(1000, GetalUtil.alsArabischCijfer("M"));
	    }
	
	    [TestMethod]
	    public void optellendeRomeinseCijfers() {
		    Assert.AreEqual(2, GetalUtil.alsArabischCijfer("II"));
		    Assert.AreEqual(3, GetalUtil.alsArabischCijfer("III"));
		    Assert.AreEqual(6, GetalUtil.alsArabischCijfer("VI"));
		    Assert.AreEqual(7, GetalUtil.alsArabischCijfer("VII"));
		    Assert.AreEqual(8, GetalUtil.alsArabischCijfer("VIII"));
		    Assert.AreEqual(11, GetalUtil.alsArabischCijfer("XI"));
		    Assert.AreEqual(17, GetalUtil.alsArabischCijfer("XVII"));
		    Assert.AreEqual(38, GetalUtil.alsArabischCijfer("XXXVIII"));
		    Assert.AreEqual(72, GetalUtil.alsArabischCijfer("LXXII"));
		    Assert.AreEqual(80, GetalUtil.alsArabischCijfer("LXXX"));
		    Assert.AreEqual(122, GetalUtil.alsArabischCijfer("CXXII"));
		    Assert.AreEqual(153, GetalUtil.alsArabischCijfer("CLIII"));
		    Assert.AreEqual(678, GetalUtil.alsArabischCijfer("DCLXXVIII"));
		    Assert.AreEqual(1233, GetalUtil.alsArabischCijfer("MCCXXXIII"));
		    Assert.AreEqual(3272, GetalUtil.alsArabischCijfer("MMMCCLXXII"));
	    }

        [TestMethod]
	    public void aftrekkendeRomeinseCijfers() {
		    Assert.AreEqual(4, GetalUtil.alsArabischCijfer("IV"));
		    Assert.AreEqual(9, GetalUtil.alsArabischCijfer("IX"));
		    Assert.AreEqual(39, GetalUtil.alsArabischCijfer("XXXIX"));
		    Assert.AreEqual(42, GetalUtil.alsArabischCijfer("XLII"));
		    Assert.AreEqual(99, GetalUtil.alsArabischCijfer("XCIX"));
		    Assert.AreEqual(109, GetalUtil.alsArabischCijfer("CIX"));
		    Assert.AreEqual(119, GetalUtil.alsArabischCijfer("CXIX"));
		    Assert.AreEqual(129, GetalUtil.alsArabischCijfer("CXXIX"));
		    Assert.AreEqual(139, GetalUtil.alsArabischCijfer("CXXXIX"));
		    Assert.AreEqual(169, GetalUtil.alsArabischCijfer("CLXIX"));
		    Assert.AreEqual(979, GetalUtil.alsArabischCijfer("CMLXXIX"));
		    Assert.AreEqual(1999, GetalUtil.alsArabischCijfer("MCMXCIX"));
	    }
    }
}

File: workspace/2_SchrijfRomeinsGetalInArabischeCijfers/test/num/SchrijfInArabischeCijfersTest.java
package num;

import static junit.framework.Assert.assertEquals;

import org.junit.Test;


public class SchrijfInArabischeCijfersTest {
	
	@Test
	public void individueleRomeinseGetallen() {
		assertEquals(1, GetalUtil.alsArabischCijfer("I"));
		assertEquals(5, GetalUtil.alsArabischCijfer("V"));
		assertEquals(10, GetalUtil.alsArabischCijfer("X"));
		assertEquals(50, GetalUtil.alsArabischCijfer("L"));
		assertEquals(100, GetalUtil.alsArabischCijfer("C"));
		assertEquals(500, GetalUtil.alsArabischCijfer("D"));
		assertEquals(1000, GetalUtil.alsArabischCijfer("M"));
	}
	
	@Test
	public void optellendeRomeinseCijfers() {
		assertEquals(2, GetalUtil.alsArabischCijfer("II"));
		assertEquals(3, GetalUtil.alsArabischCijfer("III"));
		assertEquals(6, GetalUtil.alsArabischCijfer("VI"));
		assertEquals(7, GetalUtil.alsArabischCijfer("VII"));
		assertEquals(8, GetalUtil.alsArabischCijfer("VIII"));
		assertEquals(11, GetalUtil.alsArabischCijfer("XI"));
		assertEquals(17, GetalUtil.alsArabischCijfer("XVII"));
		assertEquals(38, GetalUtil.alsArabischCijfer("XXXVIII"));
		assertEquals(72, GetalUtil.alsArabischCijfer("LXXII"));
		assertEquals(80, GetalUtil.alsArabischCijfer("LXXX"));
		assertEquals(122, GetalUtil.alsArabischCijfer("CXXII"));
		assertEquals(153, GetalUtil.alsArabischCijfer("CLIII"));
		assertEquals(678, GetalUtil.alsArabischCijfer("DCLXXVIII"));
		assertEquals(1233, GetalUtil.alsArabischCijfer("MCCXXXIII"));
		assertEquals(3272, GetalUtil.alsArabischCijfer("MMMCCLXXII"));
	}
	
	@Test
	public void aftrekkendeRomeinseCijfers() {
		assertEquals(4, GetalUtil.alsArabischCijfer("IV"));
		assertEquals(9, GetalUtil.alsArabischCijfer("IX"));
		assertEquals(39, GetalUtil.alsArabischCijfer("XXXIX"));
		assertEquals(42, GetalUtil.alsArabischCijfer("XLII"));
		assertEquals(99, GetalUtil.alsArabischCijfer("XCIX"));
		assertEquals(109, GetalUtil.alsArabischCijfer("CIX"));
		assertEquals(119, GetalUtil.alsArabischCijfer("CXIX"));
		assertEquals(129, GetalUtil.alsArabischCijfer("CXXIX"));
		assertEquals(139, GetalUtil.alsArabischCijfer("CXXXIX"));
		assertEquals(169, GetalUtil.alsArabischCijfer("CLXIX"));
		assertEquals(979, GetalUtil.alsArabischCijfer("CMLXXIX"));
		assertEquals(1999, GetalUtil.alsArabischCijfer("MCMXCIX"));
	}
	
}
